src/app_task.py

```diff
@@ -0,0 +1,15 @@
+from decode_function import BucketReader
+from mail_client import MailService
+from dynamo_client import DynamoClient
+import json
+from datetime import datetime
+
+
+def send_email(data=dict, reader=BucketReader(), mailer=MailService(), dynamo=DynamoClient(table="DailyTable")):

```

Comment:
You could move this inside a class that you initialize with the BucketReader, MailService and DynamoClient once and then call the send_email function only with the bucket, address and content. Also, don't send a dict as input for the function, it makes the code hard to read and understand.